pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -267,10 +270,30 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		})
 	})
 	registerHandlers(c.pods.informer, c.queue, "Pods", c.pods.onEvent, nil)
-
+	if err := updateSPIFFERegistry(kubeClient, c.clusterID); err != nil {
```

Comment:
Dumb question: Is there a controller for each remote cluster? Seeing this is the only way to update the trustDomainInfoMap in spiffe.

Reply:
Yes. In multi cluster case, it creates one kube controller per cluster. This includes both remote and local cluster.